src/main/java/com/hillel/items_exchange/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.hillel.items_exchange.service.impl;
+
+import com.hillel.items_exchange.dao.CategoryRepository;
+import com.hillel.items_exchange.dto.CategoryDto;
+import com.hillel.items_exchange.dto.SubcategoryDto;
+import com.hillel.items_exchange.model.Category;
+import com.hillel.items_exchange.model.Subcategory;
+import com.hillel.items_exchange.service.CategoryService;
+import com.hillel.items_exchange.service.SubcategoryService;
+import java.util.*;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import static com.hillel.items_exchange.mapper.UtilMapper.*;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final SubcategoryService subcategoryService;
+    private final ModelMapper modelMapper;
```

Comment:
You could continued to use `convertTo(T src, Class<K> kClass)` and not inject `ModelMapper` because it calls `map(Object source, Class<D> destinationType)` inside

Reply:
Great!